File: prime/golang_prime/prime.go
package main

import "os"
import "fmt"

func isPrime(number  int) bool {
    idx_i := 0
    for idx_i=2; idx_i < number; idx_i++ {
        if (number % idx_i == 0 && idx_i != number) {
          return false;
        }
    }
    return true;
}

func main() {
    args := os.Args
    argc := len(args)

    if argc != 1 {
      fmt.Println(args[0] + " <noargs>")
      os.Exit(1)
    }

    for idx_i:=0; idx_i < 99; idx_i++ {
      fmt.Printf("isPrime(%d)=<%v>\n", idx_i, isPrime(idx_i));
    }
}

File: prime/c_prime/prime.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int isPrime( unsigned long );

int main( int argc, char *argv[] ) {
  unsigned long idx_i = 0L;

  if( argc != 1 ) {
    fprintf( stderr, "Usage: %s <noargs>\n", argv[0]);
    exit( 1 );
  }
  for( idx_i = 0; idx_i < 100; idx_i++ ) {
    printf("prime %lu = %d\n", idx_i, isPrime(idx_i));
  }
  return 0;
}

int isPrime(unsigned long number) {
  unsigned long idx_i = 0L;

  for (idx_i = 2; idx_i < number; idx_i++ ) {
    if( number % idx_i == 0 && idx_i != number ) {
      return 0;
    }
  }
  return 1;
}
